packages/mattermost-redux/src/selectors/entities/channel_categories.ts

```diff
@@ -334,25 +333,12 @@ export function makeSortChannelsByNameWithDMs(): (state: GlobalState, channels:
 export function makeSortChannelsByRecency(): (state: GlobalState, channels: Channel[]) => Channel[] {
     return createSelector(
         'makeSortChannelsByRecency',
-        (state: GlobalState, channels: Channel[]) => channels,
-        getLastPostPerChannel,
+        (_state: GlobalState, channels: Channel[]) => channels,
         isCollapsedThreadsEnabled,
-        (channels, lastPosts, crtEnabled) => {
+        (channels, crtEnabled) => {
             return [...channels].sort((a, b) => {
-                // If available, get the last post time from the loaded posts for the channel, but fall back to the
-                // channel's last_post_at if that's not available. The last post time from the loaded posts is more
-                // accurate because channel.last_post_at is not updated on the client as new messages come in.
```

Comment:
Hey @koox00, can you look at this `getLastPostPerChannel` part? I removed it here and it works well when new messages come. Maybe that is possible that I missed some edge cases that this approach fixes them before. I'd like to get your opinion here. I can add this again if you prefer that.

Reply:
He's out this week, so hopefully I can answer any questions here since I wrote a bunch of this code.

I thought this comment still applied, but it turns out that we actually started updating the `channel.last_post_at` around the same time I wrote that (see https://github.com/mattermost/mattermost-redux/pull/1165 that started updating it). This should be good to remove then.

Also, I guess that explains why this might have happened then. The latest version of the logic to update the `last_post_at` takes CRT into account properly, but `getLastPostPerChannel` doesn't which lead to us not following CRT here

Reply:
That's all to say, nice catch! This looks right to me now

Reply:
Thanks for your response. That's cool! After rebasing it, I think it will be ready for review. Thank you!